mail-filter/opendkim/files/opendkim.init.r6

```diff
@@ -59,8 +59,8 @@ start_pre() {
 	# the opendkim(8) man page says).
 	if [ "${OPENDKIM_SOCKET#local:}" != "${OPENDKIM_SOCKET}" ]; then
 		# The socket begins with "local:"
-		OPENDKIM_SOCKET_PATH="${OPENDKIM_SOCKET#local:}"
-		OPENDKIM_SOCKET_DIR="${OPENDKIM_SOCKET_PATH%/*}"
+		local OPENDKIM_SOCKET_PATH="${OPENDKIM_SOCKET#local:}"
```

Comment:
OpenRC uses `/bin/sh` for its service scripts and not bash, so `local` might not be available.

Reply:
Interesting. In past pull requests, M. Górny was cross with me for not using `local` with every function body variable. Also:
```
# ls -ld /bin/sh
lrwxrwxrwx 1 root root 4 Jan 17 00:30 /bin/sh -> bash*
# equery b /bin/sh
 * Searching for /bin/sh ...
app-shells/bash-4.4_p23-r1 (/bin/bash)
# equery u app-shells/sh
!!! No packages matching 'app-shells/sh'
```
I thought this meant that Gentoo does not have an actual generic `sh`?

Reply:
That equery shows that bash owns /bin/bash (as expected); however, users may change /bin/sh to be something else that is POSIX compliant (eg dash).  @orlitzky is correct that `local` may not be available in those instances, as it is not specified in POSIX, and thus cannot be used in OpenRC initscripts intended to be portable (which would be applicable to initscripts in the Gentoo tree).

Reply:
Most shell stuff in Gentoo (that is, everything in the [PMS](https://dev.gentoo.org/~ulm/pms/head/pms.html)) is bash, and the EAPIs say exactly what version of bash. So when working on ebuilds and eclasses, you do want to use `local` everywhere you can.

OpenRC is however in theory cross-platform, and can be used on systems without bash. Instead it uses the `/bin/sh` that is guaranteed to exist by POSIX, and as a result, it can't (reliably) use any features of that shell that POSIX doesn't guarantee. Even on Gentoo, the `/bin/sh` symlink is controlled by [app-eselect/eselect-sh](https://packages.gentoo.org/packages/app-eselect/eselect-sh), so you never know what you're going to find there.

Reply:
Thanks for the explanation, to both of you.